leakcanary-android-release/src/main/java/leakcanary/MinimumDiskSpaceCondition.kt

```diff
@@ -0,0 +1,46 @@
+package leakcanary
+
+import android.app.Application
+import android.os.Build.VERSION.SDK_INT
+import android.os.StatFs
+import leakcanary.HeapAnalysisCondition.Result.StartAnalysis
+import leakcanary.HeapAnalysisCondition.Result.StopAnalysis
+import leakcanary.internal.uiHandler
+import java.util.concurrent.TimeUnit
+
+class MinimumDiskSpaceCondition(
+  private val application: Application,
+  private val minimumDiskSpaceBytes: Long = 200_000_000,
+  private val tryAgainDelayMillis: Long = TimeUnit.HOURS.toMillis(1)
+) : HeapAnalysisCondition() {
+
+  private val postedRetry = Runnable {
+    trigger.conditionChanged(
+        "$tryAgainDelayMillis ms passed since last memory check"
+    )
+  }
+
+  override fun evaluate(): Result {
+    uiHandler.removeCallbacks(postedRetry)
+    val availableDiskSpace = availableDiskSpace()
+    return if (availableDiskSpace >= minimumDiskSpaceBytes) {
+      StartAnalysis
+    } else {
+      uiHandler.postDelayed(postedRetry, tryAgainDelayMillis)
+      StopAnalysis(
+          "availableDiskSpace $availableDiskSpace < minimumDiskSpaceBytes $minimumDiskSpaceBytes"
+      )
+    }
+  }
+
+  private fun availableDiskSpace(): Long {
+    val filesDir = application.filesDir!!
```

Comment:
How safe is this `!!` here? E.g. emulators, custom hardware and firmware can potentially return `null` here?

Reply:
I hope not, that's where apps stores all of their files.